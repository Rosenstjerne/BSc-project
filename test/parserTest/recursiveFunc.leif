var int input, output;
function int factorial(int n) {
  if (n <= 1) {
    return 1;
  }
  return n * factorial(n - 1);
}

input = 5;
output = factorial(input);
print(output);