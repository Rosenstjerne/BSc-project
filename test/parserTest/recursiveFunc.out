function null main() {
	var int input, output;
	
	function int factorial(int n) {
		if((n <= 1))
		#begin if
			return 1;
			
		#end if
		return (n * factorial((n - 1)));
		
	}
	
	input = 5;
	output = factorial(input);
	print(output);
	
}
